pandas/core/arrays/datetimes.py

```diff
@@ -793,17 +793,24 @@ def month_name(self, locale=None):
         """
         Return the month names of the DateTimeIndex with specified locale.
 
+        .. versionadded:: 0.23.0
+
         Parameters
         ----------
-        locale : string, default None (English locale)
-            locale determining the language in which to return the month name
+        locale : str, optional
+            Locale determining the language in which to return the month name.
+            Default is English locale.
```

Comment:
Where is pandas making the default English? Haven't looked in detail but assumed this would be system dependent

Reply:
I would expect that to be system independent (when programming, I'd rather have predictable results).

Code is implemented in `get_date_name_field()`:

https://github.com/pandas-dev/pandas/blob/86e8f23be6d8496cb39ee836b5b02f5c91fda0ba/pandas/_libs/tslibs/fields.pyx#L102-L103
https://github.com/pandas-dev/pandas/blob/86e8f23be6d8496cb39ee836b5b02f5c91fda0ba/pandas/_libs/tslibs/fields.pyx#L117-L118

`DAYS_FULL` and `MONTHS_FULL` are hard-coded in Pandas:

https://github.com/pandas-dev/pandas/blob/86e8f23be6d8496cb39ee836b5b02f5c91fda0ba/pandas/_libs/tslibs/ccalendar.pyx#L39-L48

Reply:
The idea was for `day_name` to replace the depreciated `weekday_name` which defaulted to hardcoded english names (without localization support).

Reply:
agree this should say the default is based on the locale

Reply:
@jreback As you can see in the code snippets above, it is not based on the locale. The default is always English.

Reply:
sure it does, just pass a locale name

Reply:
@jreback Sorry, I do not understand your last comment. Are you still saying when no locale is passed as parameter then the default behavior is to use the system locale instead of always English? Or are you saying the default locale (English) should be specified instead of `locale=None` in the method declaration?

I am confused because it seems you first asked to change the docstring to state that the default was the system locale instead of English (which is not true).

Reply:
why don't you try both ways and see

Reply:
@jreback Which both ways? Could you be more explicit about what you are trying to say? Explicit is better than implicit and words are free. :stuck_out_tongue_winking_eye: 

I tried changing the system locale and verified the default behavior was to use English. Maybe I did something wrong, in which case could you provide a better way to verify it? Also, I searched through the Pandas code and it seemed to me (as shown above), that the default is English, not system locale.

Reply:
@jreback Yeah, the tests seem to verify the same behavior: when locale is `None`, the default is always English, not system locale. I still do not get what you are trying to tell me then... :cry:

Reply:
can you show here what you are running exactly

Reply:
@jreback I do not think the code I am running is relevant here. Unless you refer to a trivial example like the one I added in the docstring (but why would you need that?):

```python
idx = pd.DatetimeIndex(start='2018-09-02', freq='D', periods=3)
idx.day_name()
```

Can you explain why you still say the default when `locale=None` is based on the system's locale? I checked the tests, the code and the behavior and everything seems to prove otherwise: the default when `locale=None` is English, and hence the proposed docstring. Did you check anything?

Reply:
@jreback If you tell me exactly what changes to the current docstring you have on your head, I may better understand what you are saying (it seems other reviewers are okay with it). :heart:

Reply:
@Peque @jreback The default locale is English as `day_name` is meant to replace `weekday_name` which defaulted to english. So the docstring here is correct.

@Peque feel free to open up a new issue suggesting that `day_name` and `month_name` should default to system locale. This would be an API change.

Reply:
@mroeschke Yeah, but I do prefer the current behavior, so I would rather not open that issue (and note I am not a native English speaker nor live in an English country).

When programming, I prefer everything in English by default. :stuck_out_tongue_winking_eye:

Reply:
Gotcha @Peque. I hold a similar opinion for API transition sake.

Therefore, I think there's no action with this docstring here as far as I'm concerned.